resources/protos/commit.proto

```diff
@@ -1,22 +1,33 @@
-syntax = "proto2";
+// Copyright 2017 Sourcerer, Inc. All Rights Reserved.
 
-package tutorial;
+syntax = "proto3";
+
+package app;
 
 option java_package = "app";
 option java_outer_classname = "CommitProtos";
 
-message Flag {
-    string type;
-    string value;
+// Types of information that can be extracted from commits.
+enum EnumLabelTypes {
+    LOC_ADDED = 0; // Lines of code added.
+    LOC_DELETED = 1; // Lines of code deleted.
+    QUALITY = 2; // Quality commit.
+    LANGUAGE = 3; // Programming languages used.
+    TECHNOLOGY = 4; // Technologies used.
 }
 
-message Commit {
-    optional int32 id;
-    optional string email;
-
-    required string name = 1;
-    required int32 id = 2;
-    optional string email = 3;
+// Label is a specific extracted information about commit.
+message Label {
+    int32 type = 1; // Type of extracted information.
```

Comment:
I think you are overgeneralizing. The approach you take makes it difficult working with this info. You will get something like "if type == loc_deleted do A, elif type == language do B". It's inefficient and does not serve any purpose. Please use dedicated fields for each attribute. E.g. 

```
message CommitStats {
  int32 num_lines_add = 1;
  int32 num_lines_deleted = 2;
  ...
}
```

When you use a structure like this your parsing code becomes non-existent.